### ISSUE TITLE
imshow alpha interpolation looks wrong

### ISSUE THREAD

**Issue by fargiolas** [matplotlib/matplotlib#11316]
Hi, not much of an issue, but it seems that `imshow` is not interpolating RGBA images in the proper way. There was some discussion about a similar issue in #9906 which ended up on a documentation note:

> imshow expects RGB images adopting the straight (unassociated) alpha representation.

If that's the case `imshow` should still switch to associated colors (premultiply alpha) before interpolation to avoid artefacts like the one below, shouldn't it?

```
import matplotlib.pyplot as plt
import numpy as np

fig, (axl, axr) = plt.subplots(1, 2)

# full green image
img = np.zeros((5, 5, 4))
img[...,1] = np.ones((5, 5))
# transparent above main diagonal
img[...,3] = np.tril(np.ones((5, 5), dtype=np.uint8))

axl.imshow(img, interpolation="none")
axr.imshow(img, interpolation="bilinear")

plt.show()
```

![bug](https://user-images.githubusercontent.com/133750/40577662-a7ca9184-6109-11e8-99c6-3201b98c8c78.png)

**Comment by fargiolas** [comment]
it seems something `Agg` specific, it doesn't happen e.g. with the `cairo` backend

**Comment by tacaswell** [comment]
It looks like it is compositing it into black (0, 0, 0) rather than white (1, 1, 1)....

**Comment by fargiolas** [comment]
@tacaswell maybe my description can be a bit misleading. Black fringing is a common artifact when some part of a graphic pipeline uses premultiplied alpha while others don't, but I'm not entirely sure that's what's happening here.

**Comment by dabana** [comment]
@tacaswell was right about doing the alpha and rgb interpolation separatly.

![a_plot](https://user-images.githubusercontent.com/29574757/45233393-9c348800-b2a0-11e8-8574-6c3e4a8a7c79.jpg)

**Comment by dabana** [comment]
While building a test for this issue, I came across a problem with the *interpolation* keyword when **savefig()** is used with the SVG format. See issue #12065.

**Comment by tacaswell** [comment]
This was fixed by #12062 / #13559. Thanks @dabana !
